iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp

```diff
@@ -176,6 +176,18 @@ void CachedConnection::SyncAttachDbs() {
 // @bsimethod
 //---------------------------------------------------------------------------------------
 void CachedConnection::Execute(std::function<void(QueryAdaptorCache&,RunnableRequestBase&)> cb, std::unique_ptr<RunnableRequestBase> request) {
+    if (m_db.IsDbOpen()) {
+        // Check if the primary file data version has changed
+        uint32_t dataVersion;
+        if (BE_SQLITE_OK == GetPrimaryDb().GetFileDataVersion(dataVersion)) {
+            if (dataVersion != m_primaryFileDataVer) {
+                m_db.ClearECDbCache();

```

Comment:
This fix looks correct to me, however I'm a bit worried about performance regression. Clearing the cache is not always cheap, for example it calls `ConcurrentQueryMgr::Shutdown` and uses a Mutex and triggers OnBefore/OnAfterClear events, god knows what this all entails. We should monitor if this affects performance, and maybe put some tracker here that helps us understand how often we clear the caches now.